networkx/drawing/nx_pylab.py

```diff
@@ -448,20 +448,25 @@ def draw_networkx_nodes(
         node_color = apply_alpha(node_color, alpha, nodelist, cmap, vmin, vmax)
         alpha = None
 
-    node_collection = ax.scatter(
-        xy[:, 0],
-        xy[:, 1],
-        s=node_size,
-        c=node_color,
-        marker=node_shape,
-        cmap=cmap,
-        vmin=vmin,
-        vmax=vmax,
-        alpha=alpha,
-        linewidths=linewidths,
-        edgecolors=edgecolors,
-        label=label,
-    )
+    # Convert to one element np.array if needed
+    if not isinstance(node_shape, np.ndarray) and not isinstance(node_shape, list):
+        node_shape = np.asarray([node_shape for _ in range(len(nodelist))])
```

Comment:
My understanding is that `np.array` converts the input to an array with a copy and `np.asarray` converts without a copy if it can. Here we input a list so they both create a new array. So the preferred form is `np.array`. 
```suggestion
        node_shape = np.asarray([node_shape for _ in range(len(nodelist))])
```

Reply:
Hmm, interesting. I did not know this bit of numpy-fu.